### ISSUE TITLE
Run patchShebangs function also in patchPhase

### ISSUE THREAD

**Issue by iElectric** [NixOS/nixpkgs#5368]
- make sure that it can be disabled with existing option
- update documentation

Would also help with https://github.com/NixOS/nixpkgs/issues/183

**Comment by iElectric** [comment]
Not as trivial as I thought. This introduces more runtime dependencies that we saw before resulting into errors such as:

```
building path(s) ‘/nix/store/4adc4crfnm3bhrq3n1vr7rxz7dqmnlhw-stdenv’
output (‘/nix/store/4adc4crfnm3bhrq3n1vr7rxz7dqmnlhw-stdenv’) is not allowed to refer to path ‘/nix/store/9q76d0pr9k04rwn5nxh47w075m5fg9ld-bootstrap-tools’
note: keeping build directory ‘/tmp/nix-build-stdenv.drv-1’
cannot build derivation ‘/nix/store/ihlajjngzdqf9y4adw90ha7ilw438dar-nixpkgs-tarball-15.05pre1234.abcdef.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/ihlajjngzdqf9y4adw90ha7ilw438dar-nixpkgs-tarball-15.05pre1234.abcdef.drv’ failed
```

**Comment by lethalman** [comment]
Disable preconfigure shebang patching for every derivation that has `allowedReferences`

**Comment by domenkozar** [comment]
Yeah this won't make it into 15.05.

**Comment by Profpatsch** [comment]
Huh, this is an old one. I don’t exactly understand the motivation of running it multiple times instead of only in the fixupPhase, which is the last phase.

**Comment by matthewbauer** [comment]
This should be doable with #43833 + ```strictDeps = true````. I will try to look into it after we have those taken care of.

**Comment by dezgeg** [comment]
Hm, doing this sounds risky for cross because the build system might be copying some scripts verbatim to `$out` and those will now contain paths to buildPlatform packages.

Maybe we could just close the issue, given that the status quo has worked fine for four years since the opening of this ticket?

**Comment by matthewbauer** [comment]
Yeah, maybe that would be easiest. I might still want to make something like ```patchShebangsBuildHook``` or something, but it probably wouldn't be added by default.

**Comment by dezgeg** [comment]
I'm not really convinced that's necessary either... it's not like

````
prePatch = ''
   patchShebangs .
'';
````

is that much effort to write.over `nativeBuildInputs = [patchShebangsBuildHook];`
